### PR TITLE
Code Quality: Introduced AppThemeModeService

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

This is one of the efforts to migrate static Helpers to singleton Services.

## Task Checklist

- [x] Introduced `AppThemeModeService` : `IAppThemeModeService`
- [x] Removed SettingsViewModel, which had theme changed event handler and has been marked as obslete.

## Steps To Validate Changes

1. Launch the app
2. See the sections you have expanded are expanded expectedly
3. Go to Settings dialog
4. Change theme mode
5. See theme will be changed both in main app window and dialog

## Known Issues

_N/A_

## PR Checklist

- [x] **Development**: Related: #13490
- [ ] **Approval**: Have discussed with and got approval from the team[^1]
- [x] **Tests**: Tested accessibility on the local end
- [x] **Deployment**: Deployed the app on the local end
- [ ] **Localization**: Modified en-US string resources[^2]
- [ ] **Co-authors**: _N/A_

## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.

### CODE REVIEW

**Comment by 0x5bfa** [comment]
I executed tests on my end and it reported as 3 succeeded, 0 failed.
I couldn't figure out why the tests have being failed.

File: src/Files.Core/Services/IAppThemeModeService.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Core.Services
+{
+	public interface IAppThemeModeService
```
Comment (yaira2) [inline src/Files.Core/Services/IAppThemeModeService.cs:6]: Do you think we can simplify this to `IThemeService`?

File: src/Files.App/ViewModels/SettingsViewModel.cs
Comment (yaira2) [inline src/Files.App/ViewModels/SettingsViewModel.cs:1]: This is a nice milestone!

Comment (0x5bfa) [reply src/Files.App/ViewModels/SettingsViewModel.cs:1]: I searched references of LocalSettings namespace, which is deprecated on Files codebase, and I think all of them can be moved to our own json settings scheme. That's out of scope of this PR though.

File: src/Files.Core/Services/IAppThemeModeService.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.Core/Services/IAppThemeModeService.cs:6]: There's already AppThemeBackground and App would be necessary for such application, which have a lot of features to differentiate others.

image

Comment (0x5bfa) [reply src/Files.Core/Services/IAppThemeModeService.cs:6]: This is still on planning stage but I'm planning rename ResourcesService, JumpListService, UpdateService to have App prefix, move into Files.App.Services.Application.*.

image

This is also on planning stage but I'm planning to rename those 2 helpers (DragZoneHelper & JumpListHelper) to have App prefix to be consistent
image

I have some outlook for the future of improvements on Files codebase, so I'd like to keep as it is.

File: src/Files.App/Dialogs/PropertiesDialog.xaml
Comment (yaira2) [inline src/Files.App/Dialogs/PropertiesDialog.xaml:1]: Why was this removed?

File: src/Files.Core/Data/Enums/AppThemeMode.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Core.Data.Enums
+{
+	public enum AppThemeMode
+	{
+		Default,
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/AppThemeMode.cs:8]: Should this be "System"?

File: src/Files.Core/Services/IAppThemeModeService.cs
Comment (yaira2) [inline src/Files.Core/Services/IAppThemeModeService.cs:1]: Please add comments

File: src/Files.App/Dialogs/PropertiesDialog.xaml
Comment (0x5bfa) [reply src/Files.App/Dialogs/PropertiesDialog.xaml:1]: This was used from nowhere and was a leftover from UWP. Should I remove in another PR?

File: src/Files.Core/Data/Enums/AppThemeMode.cs (hunk 2 above)
Comment (0x5bfa) [reply src/Files.Core/Data/Enums/AppThemeMode.cs:8]: I don't have preferences. Would like to rename to System?

Comment (yaira2) [reply src/Files.Core/Data/Enums/AppThemeMode.cs:8]: As long as we keep the existing user settings
